qemu/tests/hugepage_reset.py

```diff
@@ -27,14 +28,56 @@ def run(test, params, env):
     def set_hugepage():
         """Set nr_hugepages"""
-        for h_size in (origin_nr - 2, origin_nr + 2):
-            hp_config.target_hugepages = h_size
-            hp_config.set_hugepages()
-            if params.get('on_numa_node'):
-                logging.info('Set hugepage size %s to target node %s',
-                             h_size, target_node)
-                hp_config.set_node_num_huge_pages(h_size, target_node,
-                                                  hugepage_size)
+        try:
+            for h_size in (origin_nr - 2, origin_nr + 2):
+                hp_config.target_hugepages = h_size
+                hp_config.set_hugepages()
+                if params.get('on_numa_node'):
+                    logging.info('Set hugepage size %s to target node %s',
+                                 h_size, target_node)
+                    hp_config.set_node_num_huge_pages(h_size, target_node,
+                                                      hugepage_size)
+        except ValueError as err:
+            test.cancel(err)
+
+    def run_stress():
+        def heavyload_install():
+            if session.cmd_status(test_install_cmd) != 0:
+                logging.warning("Could not find installed heavyload in guest, "
+                                "will install it via winutils.iso ")
+                winutil_drive = utils_misc.get_winutils_vol(session)
+                if not winutil_drive:
+                    test.cancel("WIN_UTILS CDROM not found.")
+                install_cmd = params["install_cmd"] % winutil_drive
+                session.cmd(install_cmd)
+
+        logging.info('Loading stress on guest.')
+        stress_duration = params.get("stress_duration", 60)
+        if params["os_type"] == "linux":
+            params['stress_args'] = '--vm %s --vm-bytes 256M --timeout %s' % (
+                    mem // 512, stress_duration)
+            stress = utils_test.VMStress(vm, 'stress', params)
+            stress.load_stress_tool()
+            time.sleep(stress_duration)
+            stress.unload_stress()
+        else:
+            session = vm.wait_for_login()
+            install_path = params["install_path"]
+            test_install_cmd = 'dir "%s" | findstr /I heavyload' % install_path
+            heavyload_install()
+            heavyload_bin = r'"%s\heavyload.exe" ' % install_path
+            heavyload_options = ["/MEMORY 100",
+                                 "/DURATION %d" % (stress_duration // 60),
+                                 "/START"]
+            start_cmd = heavyload_bin + " ".join(heavyload_options)
+            stress_tool = BackgroundTest(session.cmd, (start_cmd,
+                                                       stress_duration,
+                                                       stress_duration))
+            stress_tool.start()
+            if not utils_misc.wait_for(stress_tool.is_alive, stress_duration):
+                test.error("Failed to start heavyload process")
```

Comment:
I think, use `stress_duration` to check if stress tool run up is not suitable, it is too long.